openvino/mt/cmake/parallel.cmake

```diff
@@ -2,6 +2,111 @@
+        RESOLVE_DEPENDENCY(OMP
+                ARCHIVE_WIN "iomp.zip"
+                TARGET_PATH "${TEMP}/omp"
+                ENVIRONMENT "OMP"
+                VERSION_REGEX ".*_([a-z]*_([a-z0-9]+\\.)*[0-9]+).*")
+    elseif(LINUX AND X86_64)
+        RESOLVE_DEPENDENCY(OMP
+                ARCHIVE_LIN "iomp.tgz"
+                TARGET_PATH "${TEMP}/omp"
+                ENVIRONMENT "OMP"
+                VERSION_REGEX ".*_([a-z]*_([a-z0-9]+\\.)*[0-9]+).*")
+    elseif(APPLE AND X86_64)
+        RESOLVE_DEPENDENCY(OMP
+                ARCHIVE_MAC "iomp_20190130_mac.tgz"
+                TARGET_PATH "${TEMP}/omp"
+                ENVIRONMENT "OMP"
+                VERSION_REGEX ".*_([a-z]*_([a-z0-9]+\\.)*[0-9]+).*")
+    else()
+        message(FATAL_ERROR "Intel OMP is not available on current platform")
+    endif()
+    update_deps_cache(OMP "${OMP}" "Path to OMP root folder")
+    log_rpath_from_dir(OMP "${OMP}/lib")
+    debug_message(STATUS "intel_omp=" ${OMP})
+endif ()
+
+## TBB package
+if (THREADING STREQUAL "TBB" OR THREADING STREQUAL "TBB_AUTO")
+    reset_deps_cache(TBBROOT)
+
+    if(NOT DEFINED TBB_DIR AND NOT DEFINED ENV{TBB_DIR})
+        if (WIN32 AND X86_64)
+            #TODO: add target_path to be platform specific as well, to avoid following if
+            RESOLVE_DEPENDENCY(TBB
+                    ARCHIVE_WIN "tbb2020_20200415_win.zip"
+                    TARGET_PATH "${TEMP}/tbb"
+                    ENVIRONMENT "TBBROOT")
+        elseif(ANDROID)  # Should be before LINUX due LINUX is detected as well
+            RESOLVE_DEPENDENCY(TBB
+                    ARCHIVE_ANDROID "tbb2020_20200404_android.tgz"
+                    TARGET_PATH "${TEMP}/tbb"
+                    ENVIRONMENT "TBBROOT")
+        elseif(LINUX AND X86_64)
+            RESOLVE_DEPENDENCY(TBB
+                    ARCHIVE_LIN "tbb2020_20200415_lin_strip.tgz"
+                    TARGET_PATH "${TEMP}/tbb")
+        elseif(LINUX AND AARCH64)
+            RESOLVE_DEPENDENCY(TBB
+                    ARCHIVE_LIN "keembay/tbb2020_38404_kmb.tgz"
+                    TARGET_PATH "${TEMP}/tbb_yocto"
+                    ENVIRONMENT "TBBROOT")
+        elseif(APPLE AND X86_64)
+            RESOLVE_DEPENDENCY(TBB
+                    ARCHIVE_MAC "tbb2020_20200404_mac.tgz"
+                    TARGET_PATH "${TEMP}/tbb"
+                    ENVIRONMENT "TBBROOT")
+        else()
+            message(FATAL_ERROR "TBB is not available on current platform")
+        endif()
+    else()
+        if(DEFINED TBB_DIR)
+            get_filename_component(TBB ${TBB_DIR} DIRECTORY)
+        else()
+            get_filename_component(TBB $ENV{TBB_DIR} DIRECTORY)
+        endif()
+    endif()
+
+    update_deps_cache(TBBROOT "${TBB}" "Path to TBB root folder")
+
+    if (WIN32)
+        log_rpath_from_dir(TBB "${TBB}/bin")
+    else ()
+        log_rpath_from_dir(TBB "${TBB}/lib")
+    endif ()
+    debug_message(STATUS "tbb=" ${TBB})
+endif ()
```

Comment:
looks like we should not have this code here because it's source repo only (download dependencies, etc) while  the code below works with downloaded / installed version of TBB and it's a part of OpenVINO customer package